imx_pm_device: Builds fail when CONFIG_PM_DEVICE and CONFIG_NETWORKING are enabled for MIMXRT1170_EVK (CM7)

**Describe the bug**
The build configuration fails with the following error

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? mimxrt1170_evk_cm7

**To Reproduce**
Steps to reproduce the behavior:
1. Enable the following configurations in a sample program
   CONFIG_PM=y
   CONFIG_PM_DEVICE=y
   CONFIG_NETWORKING=y

3. Run the build for mimxrt1170_evk_cm7
4. Notice the following error: 

**Logs and console output**
   ```
Building C object zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_mcux.c.obj
FAILED: zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_mcux.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.16.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE="CONFIG_FLASH_SIZE*1024" -DCACHE_MODE_WRITE_THROUGH -DCPU_MIMXRT1176DVMAA_cm7 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNDEBUG -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DXIP_BOOT_HEADER_DCD_ENABLE=1 -DXIP_BOOT_HEADER_ENABLE=1 -DXIP_EXTERNAL_FLASH -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/workdir/deps/zephyr/include -I/workdir/build/zephyr/include/generated -I/workdir/deps/zephyr/soc/arm/nxp_imx/rt -I/workdir/deps/zephyr/subsys/net/l2 -I/workdir/deps/zephyr/subsys/net/lib/dns/. -I/workdir/deps/base/include -I/workdir/deps/modules/hal/cmsis/CMSIS/Core/Include -I/workdir/deps/zephyr/modules/cmsis/. -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1176 -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1176/drivers -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1176/xip -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/caam -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/lpuart -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/gpt -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/enet -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/igpio -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/cache/armv7-m7 -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/drivers/gpc_3 -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/components/phy -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phyksz8081 -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phyar8031 -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phylan8720a -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phyrtl8211f -I/workdir/deps/modules/hal/nxp/mcux/mcux-sdk/components/phy/mdio/enet -I/workdir/deps/zephyr/modules/hal_nxp/. -I/workdir/deps/gin_utils/include/extensions -I/workdir/deps/gin_utils/include/lib -I/workdir/deps/gin_utils/include/types -I/workdir/deps/base/modules/gp/include -isystem /workdir/deps/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /workdir/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/toolchains/zephyr-sdk-0.16.4/arm-zephyr-eabi/arm-zephyr-eabi -imacros /workdir/deps/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/workdir/deps/zephyr/samples/subsys/pm/device_pm=CMAKE_SOURCE_DIR -fmacro-prefix-map=/workdir/deps/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/workdir=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_mcux.c.obj -MF zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_mcux.c.obj.d -o zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_mcux.c.obj -c /workdir/deps/zephyr/drivers/ethernet/eth_mcux.c
In file included from /workdir/deps/zephyr/include/zephyr/sys/util_macro.h:34,
                 from /workdir/deps/zephyr/include/zephyr/sys/util.h:17,
                 from /workdir/deps/zephyr/include/zephyr/sys/cbprintf_internal.h:15,
                 from /workdir/deps/zephyr/include/zephyr/sys/cbprintf.h:124,
                 from /workdir/deps/zephyr/include/zephyr/logging/log_msg.h:11,
                 from /workdir/deps/zephyr/include/zephyr/logging/log_core.h:9,
                 from /workdir/deps/zephyr/include/zephyr/logging/log.h:11,
                 from /workdir/deps/zephyr/drivers/ethernet/eth_mcux.c:23:
/workdir/deps/zephyr/include/zephyr/device.h:89:41: error: '__device_dts_ord_DT_N_S_soc_S_ethernet_40424000_P_clocks_IDX_0_PH_ORD' undeclared here (not in a function)
   89 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/devicetree.h:4112:9: note: in expansion of macro 'COND_CODE_1'
 4112 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/workdir/deps/zephyr/drivers/ethernet/eth_mcux.c:1498:9: note: in expansion of macro 'COND_CODE_1'
 1498 |         COND_CODE_1(CONFIG_NET_POWER_MANAGEMENT,                        \
      |         ^~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/toolchain/common.h:138:23: note: in expansion of macro '_DO_CONCAT'
  138 | #define _CONCAT(x, y) _DO_CONCAT(x, y)
      |                       ^~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/device.h:89:33: note: in expansion of macro '_CONCAT'
   89 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                 ^~~~~~~
/workdir/deps/zephyr/include/zephyr/device.h:216:37: note: in expansion of macro 'DEVICE_NAME_GET'
  216 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))
      |                                     ^~~~~~~~~~~~~~~
/workdir/deps/zephyr/include/zephyr/device.h:233:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  233 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/workdir/deps/zephyr/drivers/ethernet/eth_mcux.c:1495:22: note: in expansion of macro 'DEVICE_DT_GET'
 1495 |         .clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),             \
      |                      ^~~~~~~~~~~~~
/workdir/deps/zephyr/drivers/ethernet/eth_mcux.c:1499:22: note: in expansion of macro 'ETH_MCUX_POWER_INIT'
 1499 |                     (ETH_MCUX_POWER_INIT(n)),                           \
      |                      ^~~~~~~~~~~~~~~~~~~
/workdir/deps/zephyr/drivers/ethernet/eth_mcux.c:1593:17: note: in expansion of macro 'ETH_MCUX_POWER'
 1593 |                 ETH_MCUX_POWER(n)                                       \
      |                 ^~~~~~~~~~~~~~
/workdir/build/zephyr/include/generated/devicetree_generated.h:78606:55: note: in expansion of macro 'ETH_MCUX_INIT'
78606 | #define DT_FOREACH_OKAY_INST_nxp_kinetis_ethernet(fn) fn(0)
      |                                                       ^~
/workdir/deps/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_kinetis_ethernet'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/workdir/deps/zephyr/drivers/ethernet/eth_mcux.c:1652:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
 1652 | DT_INST_FOREACH_STATUS_OKAY(ETH_MCUX_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[275/285] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/ipv6_nbr.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /workdir/build
```

**Expected behavior**
The build should be successful. 

**Impact**
Cannot proceed with the device runtime power management (showstopper)

**Environment (please complete the following information):**
    OS: Linux
    Toolchain: Zephyr SDK with West
    Version used: 3.5.99

**Additional context**
For the convenience of our project, the Zephyr SDK has been moved to /workdir/deps/zephyr/...

--- a/drivers/ethernet/eth_mcux.c
+++ b/drivers/ethernet/eth_mcux.c
@@ -240,6 +240,7 @@ struct eth_context {
 	ROUND_UP(ENET_FRAME_MAX_FRAMELEN, ENET_BUFF_ALIGNMENT)
 #endif /* CONFIG_NET_VLAN */
 
+#ifdef CONFIG_SOC_FAMILY_KINETIS
 #if defined(CONFIG_NET_POWER_MANAGEMENT)
 static void eth_mcux_phy_enter_reset(struct eth_context *context);
 void eth_mcux_phy_stop(struct eth_context *context);
@@ -292,6 +293,7 @@ out:
 	return ret;
 }
 #endif /* CONFIG_NET_POWER_MANAGEMENT */
+#endif /* CONFIG_SOC_FAMILY_KINETIS */
 
 #if ETH_MCUX_FIXED_LINK
 static void eth_mcux_get_phy_params(phy_duplex_t *p_phy_duplex,
@@ -1491,6 +1493,7 @@ static void eth_mcux_err_isr(const struct device *dev)
 		    (ETH_MCUX_MAC_ADDR_LOCAL(n)),			\
 		    (ETH_MCUX_MAC_ADDR_GENERATE(n)))
 
+#ifdef CONFIG_SOC_FAMILY_KINETIS
 #define ETH_MCUX_POWER_INIT(n)						\
 	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),		\
 
@@ -1498,6 +1501,15 @@ static void eth_mcux_err_isr(const struct device *dev)
 	COND_CODE_1(CONFIG_NET_POWER_MANAGEMENT,			\
 		    (ETH_MCUX_POWER_INIT(n)),				\
 		    (ETH_MCUX_NONE))
+#define ETH_MCUX_PM_DEVICE_INIT(n)					\
+	PM_DEVICE_DT_INST_DEFINE(n, eth_mcux_device_pm_action);
+#define ETH_MCUX_PM_DEVICE_GET(n) PM_DEVICE_DT_INST_GET(n)
+#else
+#define ETH_MCUX_POWER(n)
+#define ETH_MCUX_PM_DEVICE_INIT(n)
+#define ETH_MCUX_PM_DEVICE_GET(n) NULL
+#endif /* CONFIG_SOC_FAMILY_KINETIS */
+
 #define ETH_MCUX_GEN_MAC(n)                                             \
 	COND_CODE_0(ETH_MCUX_MAC_ADDR_TO_BOOL(n),                       \
 		    (ETH_MCUX_GENERATE_MAC(n)),                         \
@@ -1629,11 +1641,11 @@ static void eth_mcux_err_isr(const struct device *dev)
 		ETH_MCUX_PTP_FRAMEINFO(n)				\
 	};								\
 									\
-	PM_DEVICE_DT_INST_DEFINE(n, eth_mcux_device_pm_action);		\
+	ETH_MCUX_PM_DEVICE_INIT(n)					\
 									\
 	ETH_NET_DEVICE_DT_INST_DEFINE(n,				\
 			    eth_init,					\
-			    PM_DEVICE_DT_INST_GET(n),			\
+			    ETH_MCUX_PM_DEVICE_GET(n),			\
 			    &eth##n##_context,				\
 			    &eth##n##_buffer_config,			\
 			    CONFIG_ETH_INIT_PRIORITY,			\

